### PR TITLE
Fix centering of full-symmetry cartesian latticeMaps

### PR DESCRIPTION
**Pull request by youngmit**
Asciimaps assume everything starts at (0,0), so grid blueprints need to
offset indices when reading in full geometry inputs. Also, when writing
asciimaps, gridContents need to be offset in the other direction for
things to work out correctly.

This also fixes a bug related to asciimaps writing cartesian maps at
all.

### CODE REVIEW

**Comment by youngmit** [comment]
Fixes #332

**Comment by jakehader** [comment]
The code changes look good. Can you post a screenshot of the before and after for the full core Cartesian case with the `gridEditor` just to show the differences?

File: armi/reactor/blueprints/tests/test_gridBlueprints.py [hunk 1]
```diff
@@ -242,7 +242,9 @@ def test_simple_read(self):
 
         gridDesign2 = self.grids["sfp"]
         _grid = gridDesign2.construct()
-        self.assertEqual(gridDesign2.gridContents[1, 1], "3")
+        self.assertEqual(gridDesign2.gridContents[1, 1], "1")
+        self.assertEqual(gridDesign2.gridContents[0, 0], "3")
+        self.assertEqual(gridDesign2.gridContents[-1, -1], "3")
```
Comment (jakehader) [inline armi/reactor/blueprints/tests/test_gridBlueprints.py:247]: Can you add unit tests to cover the following scenarios:

- Cartesian, Full Core, Even
- Cartesian, Quarter Core, Odd
- Cartesian, Quarter Core, Even

The SFP test is nice, but let's make sure that we have the other variations tested too.

**Review by jakehader — CHANGES_REQUESTED** [review]
Can we improve some of the `grids` documentation to include the images that you provided me with where the origin points are for the different grid types and symmetries? I think that this will be very useful going forward as a visual aid for users and developers.

**Comment by youngmit** [comment]
Before:
image


After:
image

**Comment by youngmit** [comment]
Also, switching to 1/4 symmetry in the blueprints works as expected:
image

File: armi/reactor/blueprints/tests/test_gridBlueprints.py (hunk 1 above)
Comment (youngmit) [reply armi/reactor/blueprints/tests/test_gridBlueprints.py:269]: Added quarter and even cases

File: armi/reactor/blueprints/tests/test_gridBlueprints.py [hunk 2]
```diff
@@ -240,12 +261,29 @@ def test_simple_read(self):
...
+        self.assertEqual(gridDesign3.gridContents[3, 3], "1")
+
+        # Cartesian full, even/odd hybrid
+        gridDesign4 = self.grids["sfp even"]
+        _grid = gridDesign4.construct()
+        self.assertEqual(gridDesign4.gridContents[0, 0], "4")
+        self.assertEqual(gridDesign4.gridContents[-1, -1], "2")
+        self.assertEqual(gridDesign4.gridContents[2, 2], "2")
+        self.assertEqual(gridDesign4.gridContents[-3, -3], "1")
+
```
Comment (jakehader) [inline armi/reactor/blueprints/tests/test_gridBlueprints.py:286]: One last comment, can you add a check for grid contents that are outside the map? I assume these should resolve to None?

**Review by jakehader — CHANGES_REQUESTED** [review]

File: armi/reactor/blueprints/tests/test_gridBlueprints.py (hunk 2 above)
Comment (youngmit) [reply armi/reactor/blueprints/tests/test_gridBlueprints.py:288]: done

**Review by jakehader — APPROVED** [review]

**Pull request merged**